pyomo/contrib/pyros/pyros.py

```diff
@@ -836,6 +247,53 @@ def _log_config(self, logger, config, exclude_options=None, **log_kwargs):
                 logger.log(msg=f" {key}={val!r}", **log_kwargs)
         logger.log(msg="-" * self._LOG_LINE_LENGTH, **log_kwargs)
 
+    def _resolve_and_validate_pyros_args(self, model, **kwds):
+        """
+        Resolve and validate arguments to ``self.solve()``.
+
+        Parameters
+        ----------
+        model : ConcreteModel
+            Deterministic model object passed to ``self.solve()``.
+        **kwds : dict
+            All other arguments to ``self.solve()``.
+
+        Returns
+        -------
+        config : ConfigDict
+            Standardized arguments.
+
+        Note
+        ----
+        This method can be broken down into three steps:
+
+        1. Resolve user arguments based on how they were passed
+           and order of precedence of the various means by which
+           they could be passed.
+        2. Cast resolved arguments to ConfigDict. Argument-wise
+           validation is performed automatically.
+        3. Inter-argument validation.
+        """
+        options_dict = kwds.pop("options", {})
+        resolved_kwds = resolve_keyword_arguments(
+            prioritized_kwargs_dicts={
+                "explicitly": kwds,
+                "implicitly through argument 'options'": options_dict,
+            },
+            func=self.solve,
+        )
+        config = self.CONFIG(resolved_kwds)
```

Comment:
This seems complex.  This could be replaced by:
```python
config = self.CONFIG(kwds.pop("options", {}))(kwds)
```
or (if you like verbosity / clarity:
```python
config = self.CONFIG(kwds.pop("options", {}))
config = config(kwds)
```